### ISSUE TITLE
Incorrect/incomplete type definition in cy.request

### ISSUE THREAD

**Issue by qubidt** [cypress-io/cypress#2305]
### Current behavior:

Regarding `cy.request(options)`:

The [type definition for `qs` in `RequestOptions`](https://github.com/cypress-io/cypress/blob/4ead424d4c9e3d65422c71597496539290720057/cli/types/index.d.ts#L1859) is `qs: string`. 

### Desired behavior:

qs should be ["an object of key/vals" that represents the query parameters in the url](https://github.com/cypress-io/cypress/issues/321). The type definition should allow key value pairs for the paramters. (Perhaps `{ [s: string]: string; }`?)

### Steps to reproduce:

Attempting to call: 
```
cy.request({
    url: `...`,
    qs: {
        some_param: "some_value"
    }
})
```

will produce a typescript error:
```
Argument of type '{ url: string; qs: { some_param: string }' is not assignable to parameter of type 'Partial<RequestOptions>'.
  Types of property 'qs' are incompatible.
    Type '{ some_param: string }' is not assignable to type 'string | undefined'.
```

### Versions

Cypress 3.0.3
typescript 3.0.1

---

I'm only guessing at the issue here, apologies if I'm fundamentally misunderstanding something. It's my first time working with any typescript/js

**Comment by bahmutov** [comment]
let me check @qubidt if there is a change, I will just add to the same PR

**Comment by brian-mann** [comment]
Released in [`3.1.4`](https://docs.cypress.io/guides/references/changelog.html#3-1-4).
